File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/Slides.java
package org.firstinspires.ftc.teamcode;

import com.qualcomm.robotcore.hardware.DcMotor;
import com.qualcomm.robotcore.hardware.DcMotorEx;
import com.qualcomm.robotcore.hardware.DcMotorSimple;
import com.qualcomm.robotcore.hardware.HardwareMap;
import com.qualcomm.robotcore.util.Range;

public class Slides {
    private DcMotorEx leftLinearSlide;
    private DcMotorEx rightLinearSlide;

    public void init(HardwareMap hardwareMap) {
        this.leftLinearSlide = hardwareMap.get(DcMotorEx.class, "7");
        this.rightLinearSlide = hardwareMap.get(DcMotorEx.class, "6");


        this.leftLinearSlide.setDirection(DcMotorSimple.Direction.REVERSE);
        this.rightLinearSlide.setDirection(DcMotorSimple.Direction.FORWARD);


        this.leftLinearSlide.setZeroPowerBehavior(DcMotor.ZeroPowerBehavior.BRAKE);
        this.rightLinearSlide.setZeroPowerBehavior(DcMotor.ZeroPowerBehavior.BRAKE);
    }

    public void slidePower(double power) {
        double clampedPower = Range.clip(power, -1, 1);

        this.leftLinearSlide.setPower(clampedPower);
        this.rightLinearSlide.setPower(clampedPower);
    }
}

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/powerplay/subsystems/Slides.kt
package org.firstinspires.ftc.teamcode.powerplay.subsystems

import com.qualcomm.robotcore.hardware.DcMotor
import com.qualcomm.robotcore.hardware.HardwareMap
import org.firstinspires.ftc.teamcode.util.Globals
import org.firstinspires.ftc.teamcode.util.Subsystem

class Slides: Subsystem {
    lateinit var slides: DcMotor

    private var slideState = SlideStates.OFF
    private enum class SlideStates(val power: Double) {
        UP(Globals.SLIDES_UP),
        DOWN(Globals.SLIDES_DOWN),
        OFF(Globals.SLIDES_OFF)
    }

    fun up() {
        slideState = SlideStates.UP
    }

    fun down() {
        slideState = SlideStates.DOWN
    }

    fun off() {
        slideState = SlideStates.OFF
    }

    override fun init(hardwareMap: HardwareMap) {
        slides = hardwareMap.dcMotor["Slides"]
    }

    override fun update() {
        slides.power = slideState.power
    }

    override fun reset() {
        off()
    }
}